### PR TITLE
Add matchDebug() for Cliqz

### PR DESCRIPTION
**Pull request by mjethani**
This patch adds a `matchDebug()` implementation for Cliqz.

### CODE REVIEW

**Comment by mjethani** [comment]
Sorry I just noticed https://github.com/cliqz-oss/adblocker/pull/2160. I have made this only for Cliqz now.

File: packages/adblocker-benchmarks/blockers/cliqz-base.js [hunk 1]
```diff
@@ -36,4 +36,12 @@ module.exports = class Cliqz {
       type,
     })).match;
   }
+
+  matchDebug({ url, frameUrl, type }) {
+    return this.engine.match(Request.fromRawDetails({
+      url,
+      sourceUrl: frameUrl,
+      type,
+    })).filter || null;
```
Comment (remusao) [inline packages/adblocker-benchmarks/blockers/cliqz-base.js:45]: By default the original filter might not be kept around (but the engine will try to recreate it from internal state. If we know we are in debug mode we can specify it to the `FiltersEngine.parse` method above:
```js

  static parse(rawLists, enableCompression, debug) {
    return new Cliqz(FiltersEngine.parse(rawLists, {
      enableCompression,
      integrityCheck: false,
      loadCosmeticFilters: false,
      debug: debug === true,
    }));
  }
```

Comment (remusao) [inline packages/adblocker-benchmarks/blockers/cliqz-base.js:45]: Also I think `filter` will be an object and not sure if the `toString()` will be called implicitly somewhere? If not:
```suggestion
    })).filter?.toString() || null;
```

Comment (mjethani) [reply packages/adblocker-benchmarks/blockers/cliqz-base.js]: Done.

**Review by remusao — APPROVED** [review]
Looks good, thank you!

**Pull request merged**